### PR TITLE
ddl: add index using ingest 640 hotfix

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: cherry-pick #39239 and #39190 to release 6.4.0-hotfix

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Benjamin2037 — APPROVED** [review]
LGTM

File: ddl/reorg.go [hunk 1]
```diff
@@ -234,7 +234,7 @@ func (w *worker) runReorgJob(rh *reorgHandler, reorgInfo *reorgInfo, tblInfo *mo
 			return dbterror.ErrCancelledDDLJob
 		}
 		rowCount, _, _ := rc.getRowCountAndKey()
-		logutil.BgLogger().Info("[ddl] run reorg job done", zap.Int64("handled rows", rowCount))
+		logutil.BgLogger().Info("[ddl] run reorg job done", zap.Int64("handled rows", rowCount), zap.Error(err))
```
Comment (hawkingrei) [inline ddl/reorg.go:237]: it contains err message. I think the log level is not info.

Comment (hawkingrei) [reply ddl/reorg.go:237]: It should be like this. 

```
if err != nil {
     log.Error(....)
} else {
     log.Info(....)
}
```

**Review by hawkingrei — APPROVED** [review]
Everything else is fine.

File: ddl/reorg.go (hunk 1 above)
Comment (tangenta) [reply ddl/reorg.go:237]: The error can be `nil`.

Comment (tangenta) [reply ddl/reorg.go:237]: OK. Let's improve it in master.

**Pull request merged**